connector/src/test/scala/com/google/cloud/spark/bigquery/it/SparkBigQueryEndToEndReadITSuite.scala

```diff
@@ -163,6 +165,32 @@ class SparkBigQueryEndToEndReadITSuite extends FunSuite
 
   def testsWithReadInFormat(dataSourceFormat: String, dataFormat: String): Unit = {
 
+    test(("testing view with different columns for select and filter. " +
+      "DataSource %s. Data Format %s").format(dataSourceFormat, dataFormat)) {
+
+      for(i <- 1 to 2){
+        var df = spark.read.format(dataSourceFormat)
+          .option("table", ALL_TYPES_VIEW_NAME)
+          .option("viewsEnabled", "true")
+          .option("viewMaterializationProject", System.getenv("GOOGLE_CLOUD_PROJECT"))
+          .option("viewMaterializationDataset", testDataset)
+          .option("readDataFormat", dataFormat)
+          .load()
+
+        if(i==2){
+          df = df.cache()
+        }
+
+        val result = df
```

Comment:
Please add comment - should this filter run on the BigQuery side or on the Spark side?

Reply:
done